# New Section

In [1]:
from keras.datasets import mnist
(train_images, train_labels), (test_images, test_labels) = mnist.load_data() # Load data

In [2]:
train_images = train_images.reshape((60000 , 28, 28, 1)) # BATCH SIZE, HEIGHT, WIDTH, CHANNELS
train_images = train_images.astype('float32')/255 # NORMALIZE DATA TO 0-1 RANGE (GREYSCALE)

test_images = test_images.reshape((10000,28,28,1))
test_images = test_images.astype('float32')/255

In [3]:
from tensorflow.keras.utils import to_categorical

train_labels = to_categorical(train_labels) # ONE-HOT ENCODE LABELS
test_labels = to_categorical(test_labels)

In [4]:
INPUT_SHAPE = (28,28,1) # HEIGHT, WIDTH, CHANNELS

In [5]:
import tensorflow as tf
model = tf.keras.models.Sequential([
    tf.keras.layers.Conv2D(64, (3,3), activation = 'relu',
                           input_shape=(28, 28, 1)),
    tf.keras.layers.MaxPooling2D(2,2),
    tf.keras.layers.Conv2D(64, (3,3), activation = 'relu'),
    tf.keras.layers.MaxPooling2D(2,2),
    tf.keras.layers.Flatten(),
    tf.keras.layers.Dense(128, activation = 'relu'),
    tf.keras.layers.Dense(10, activation= 'softmax')
])
model.compile()

In [8]:
model = Sequential()
model.add(Conv2D(32, (3,3), activation = "relu", input_shape = INPUT_SHAPE))
model.add(MaxPooling2D((2,2)))
model.add(Flatten())
model.add(Dense(10,activation = "softmax"))






IndentationError: ignored

In [6]:
model.compile(optimizer = "rmsprop" , loss = 'categorical_crossentropy', metrics = ["accuracy"])
model.fit(train_images, train_labels, batch_size = 64, epochs = 7)


Epoch 1/7
938/938 [==============================] - 90s 94ms/step - loss: 0.1440 - accuracy: 0.9549
Epoch 2/7
938/938 [==============================] - 81s 87ms/step - loss: 0.0423 - accuracy: 0.9869
Epoch 3/7
938/938 [==============================] - 81s 87ms/step - loss: 0.0281 - accuracy: 0.9912
Epoch 4/7
938/938 [==============================] - 82s 87ms/step - loss: 0.0195 - accuracy: 0.9940
Epoch 5/7
938/938 [==============================] - 81s 87ms/step - loss: 0.0146 - accuracy: 0.9960
Epoch 6/7
938/938 [==============================] - 81s 86ms/step - loss: 0.0108 - accuracy: 0.9967
Epoch 7/7
938/938 [==============================] - 87s 93ms/step - loss: 0.0089 - accuracy: 0.9973


Epoch 1/7
938/938 [==============================] - 90s 94ms/step - loss: 0.1440 - accuracy: 0.9549
Epoch 2/7
938/938 [==============================] - 81s 87ms/step - loss: 0.0423 - accuracy: 0.9869
Epoch 3/7
938/938 [==============================] - 81s 87ms/step - loss: 0.0281 - accuracy: 0.9912
Epoch 4/7
938/938 [==============================] - 82s 87ms/step - loss: 0.0195 - accuracy: 0.9940
Epoch 5/7
938/938 [==============================] - 81s 87ms/step - loss: 0.0146 - accuracy: 0.9960
Epoch 6/7
938/938 [==============================] - 81s 86ms/step - loss: 0.0108 - accuracy: 0.9967
Epoch 7/7
938/938 [==============================] - 87s 93ms/step - loss: 0.0089 - accuracy: 0.9973
<keras.src.callbacks.History at 0x7f26eac473a0>

In [7]:
model.evaluate(test_images, test_labels)

313/313 [==============================] - 5s 12ms/step - loss: 0.0296 - accuracy: 0.9928


[0.029616372659802437, 0.9927999973297119]

313/313 [==============================] - 5s 12ms/step - loss: 0.0296 - accuracy: 0.9928
[0.029616372659802437, 0.9927999973297119]